File: Homework 4/JAVA/NameAddress.java
/*
 * Objective: Creates object that includes a full name (first middle last or last, first middle) 
*			and an address (street, city, state, zip)
*/

public class NameAddress {
	
    private String street, spacedName, commaName;
	private SingleName city, state, first, middle, last;
	private int zip;

	public NameAddress(String street, SingleName city, SingleName state, 
			int zip, SingleName first, SingleName middle, SingleName last) {
		this.first = first;
		this.middle = middle;
		this.last = last;
		this.street = street;
		this.city = city;
		this.state = state;
		this.zip = zip;
	}

	public NameAddress() {
	}
	

void setStreet(String street ) {
	this.street = street;
}

String getStreet() {

	return street;
}
void setCity(SingleName city) {
	this.city = city;
}
SingleName getCity() {
	
	return city;
}

void setState(SingleName state) {
	this.state = state;
}

SingleName getState() {
	
	return state;
}

void setZip(int zip) {
	this.zip = zip;
}

int getZip() {
	
	return zip;
}

public void setFirst(SingleName first) {
	this.first = first;
}

public SingleName getFirst() {
	return first;
}

public void setMiddle(SingleName middle) {
	this.middle = middle;
}

public SingleName getMiddle() {
	return middle;
}

public void setLast(SingleName last) {
	this.last = last;
}

public SingleName getLast() {
	return last;
}

public void setAll(SingleName first, SingleName middle, SingleName last) {
	this.first = first;
	this.middle = middle;
	this.last = last;
}

/*
 * returns full name formatted with spaces 
 */
public String getFullNameSpaces() {
	spacedName = first.get() + " " + middle.get() + " " + last.get();
	return spacedName;
}

/*
 * returns full name formatted with last name, first "Smith, Bob T"
 */
public String getLastCommaFirst() {
	commaName = last.get() + ", " + first.get() + " " + middle.get();
	return commaName;
}
}

File: Homework 4/C++/NameAddress.cpp
//#include <locale>
#include "NameAddress.h"

using namespace std;

NameAddress::NameAddress()
{
	name = FullName();
	address = Address();
}

NameAddress::~NameAddress()
{
}

SingleName NameAddress::getFirst()
{
	SingleName ret = name.getFirst();
	return ret;
}

SingleName NameAddress::getMiddle()
{
	SingleName ret = name.getMiddle();
	return ret;
}

SingleName NameAddress::getLast()
{
	SingleName ret = name.getLast();
	return ret;
}

string NameAddress::getFullNameSpaces()
{
	string space = " ";
	string concat = "";
	AlphaString aFirst = name.getFirst().get();
	AlphaString aMiddle = name.getMiddle().get();
	AlphaString aLast = name.getLast().get();

	concat.append(aFirst.get());
	concat.append(space);
	concat.append(aMiddle.get());
	concat.append(space);
	concat.append(aLast.get());

	return concat;
}

string NameAddress::getLastCommaFirst()
{
	string commaSpace = ", ";
	string concat = "";
	AlphaString aFirst = name.getFirst().get();
	AlphaString aLast = name.getLast().get();

	concat.append(aLast.get());
	concat.append(commaSpace);
	concat.append(aFirst.get());

	return concat;
}

void NameAddress::setFirst(SingleName input)
{
	name.setFirst(input);
}

void NameAddress::setMiddle(SingleName input)
{
	name.setMiddle(input);
}
void NameAddress::setLast(SingleName input)
{
	name.setLast(input);
}

void NameAddress::setName(SingleName inputFirst, SingleName inputMiddle, SingleName inputLast)
{
	name.setFirst(inputFirst);
	name.setMiddle(inputMiddle);
	name.setLast(inputLast);
}

void NameAddress::setStreet(string input) {
	address.setStreet(input);
}

void NameAddress::setCity(SingleName input) {
	address.setCity(input);
}

void NameAddress::setState(SingleName input) {
	address.setState(input);
}

void NameAddress::setZip(int input) {
	address.setZip(input);
}

string NameAddress::getStreet() {
	return address.getStreet();
}

SingleName NameAddress::getCity() {
	return address.getCity();
}

SingleName NameAddress::getState() {
	return address.getState();
}

int NameAddress::getZip() {
	return address.getZip();
}